File: pset2/readability.c
# include <stdio.h>
# include <cs50.h>
# include <math.h>
# include <string.h>

int main()
{
    
    float letters = 1, words = 1, sentences = 0;
    string x = get_string("Text : ");   // Getting Input
    // printf("%s\n", x);
    
    for (int i = 1; i < strlen(x); ++i) 
    {
        if ((x[i] >= 'A' && x[i] <= 'Z') || (x[i] >= 'a' && x[i] <= 'z') || x[i] == '-')
        {
            letters++;  // Letters Count
            printf("%c",x[i]);
        }
        if (x[i] == ' ') 
        {
            ++words;    // Words Count
            if (x[i + 1] == ' ')
            {
                --words;    // Adjacent Spaces
            }
        }
        
        if (x[i] == '.' || x[i] == '!' || x[i] == '?')
        {
            ++sentences;    // Sentences Count
        }
    }
    
    float L = letters / words * 100;
    float S = sentences / words * 100;

    printf("%f\n%f\n%f\n%f\n%f\n", letters, words,sentences,L, S);
    int index = round(0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8);
    
    if (index < 0)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16) 
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

s = get_string('Text:')  # Getting Input

letters = words = 1.0
index = 0
sentences = 0.0

for i in range(1, len(s)):  # Loop through the string
    if s[i] == " ":  # Word Count
        words += 1
        if s[i + 1] == ' ':  # Adjacent Spaces Count
            words -= 1
    elif s[i] == ',' or s[i] == "'":    
        i += 1
    elif s[i].isalpha() or s[i] == '-':  # Alphabets & Hyphen Count
        letters += 1
        
sentences = s.count('.') + s.count('!') + s.count('?')  # Sentences Count
L_Per_W = (letters / words) * 100  # Counting Averages
S_Per_W = (sentences / words) * 100  # Counting Averages
X = 0.0588 * L_Per_W - 0.296 * S_Per_W - 15.8
index = round(X)
if index < 0:
    print('Before Grade 1')
elif index > 16:
    print('Grade 16+')
else:
    print("Grade {}".format(index))